src/icrar/leap-accelerate/cuda/device_matrix.h

```diff
@@ -77,28 +100,13 @@ namespace cuda
             }
         }
 
-        device_matrix(Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> data)
+        explicit device_matrix(Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> data)
         : device_matrix(data.rows(), data.cols(), data.data()) {}
 
         template<int Rows, int Cols>
-        device_matrix(Eigen::Matrix<T, Rows, Cols> data)
+        explicit device_matrix(Eigen::Matrix<T, Rows, Cols> data)
```

Comment:
Unrelated to the addition of `explicit` (which is fine): don't these two ctors actually do an unnecessary data copy? IIRC `Matrix` copy-constructs the underlying data, right? As a side effect this also makes the subsequent `cudaMemcpyAsync` call potentially unsafe.

Reply:
These are meant to be passed by reference

Reply:
I'm a little surprised I didn't get a copy performance warning here but this may result in some speedup. Tests are passing locally but I'll go through and check where I need to perform cpu wait